### PR TITLE
Factor out post-init swagger and OpenAPI routes

### PR DESCRIPTION
**Pull request by sttts**
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/32590)

### CODE REVIEW

File: pkg/genericapiserver/genericapiserver.go [hunk 1]
```diff
@@ -248,12 +243,20 @@ func (s *GenericAPIServer) installGroupsDiscoveryHandler() {
...
+	// the API group routes are created *after* initialization and you
+	// can't generate the swagger routes until after those are available.
+	// TODO eventually we should be able to factor this out to take place during initialization.
 	if s.enableSwaggerSupport {
-		s.InstallSwaggerAPI()
+		routes.SwaggerAPI{ExternalAddress: s.ExternalAddress}.Install(s.Mux, s.HandlerContainer)
 	}
 	if s.enableOpenAPISupport {
-		s.InstallOpenAPI()
+		routes.OpenAPI{
```
Comment (deads2k) [inline pkg/genericapiserver/genericapiserver.go]: I should probably know this, but what is this `OpenAPI` thing?

Comment (sttts) [inline pkg/genericapiserver/genericapiserver.go]: Swagger alternative: https://openapis.org/

**Comment by deads2k** [comment]
just a question.  lgtm

File: pkg/genericapiserver/genericapiserver.go (hunk 1 above)
Comment (sttts) [inline pkg/genericapiserver/genericapiserver.go]: ... that's what I thought. Wikipedia: "The OpenAPI Specification (originally known as the Swagger Specification)".

Comment (nikhiljindal) [inline pkg/genericapiserver/genericapiserver.go]: yes swagger 2.0 is OpenAPI spec which is designed by the Open API initiative (OAI)

**Comment by nikhiljindal** [comment]
cc @mbohlool

**Comment by sttts** [comment]
@deads2k unblocked this. ptal

**Comment by sttts** [comment]
Rebased, @deads2k ptal

**Comment by deads2k** [comment]
lgtm.  Squash tests in with the code that required their change if you would.

**Comment by sttts** [comment]
@k8s-bot gci gke e2e test this

**Comment by sttts** [comment]
@k8s-bot unit test this

**Comment by sttts** [comment]
Bazel fun.

**Pull request merged**